pkg/route/api/v1/types.go

```diff
@@ -52,6 +56,14 @@ type RouteSpec struct {
 	TLS *TLSConfig `json:"tls,omitempty"`
 }
 
+// RouteTargetReference specifies the target that resolve into endpoints. Only the 'Service'
+// kind is allowed. Use 'weight' field to emphasize one over others.
+type RouteTargetReference struct {
+	Kind string `json:"kind"`
+	Name string `json:"name"`
+	Weight int `json:"weight"`
```

Comment:
Weight needs to be a *int32, to distinguish between set and unset (end users who don't care about weight should not be getting a weight of zero, which is not backwards compatible).  Defaulting of weight needs to be added to the scheme so that weights default to a specific value if omitted (if the pointer is nil).